File: go/api/Ntwk.go
package api

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// Ntwk 对应 Ntwk.py。
type Ntwk struct {
	*session.ApiGroup
}

// NewNtwk 创建 Ntwk API 分组。
func NewNtwk(s *session.Session) *Ntwk {
	return &Ntwk{ApiGroup: session.NewApiGroup(s)}
}

// LanUpnpPortmapping 对应 ntwk/lan_upnp_portmapping。
func (n *Ntwk) LanUpnpPortmapping() (map[string]interface{}, error) {
	return n.S.Get("ntwk/lan_upnp_portmapping", nil, "api")
}

// Celllock 对应 ntwk/celllock。
func (n *Ntwk) Celllock() (map[string]interface{}, error) {
	return n.S.Get("ntwk/celllock", nil, "api")
}

// Dualwaninfo 对应 ntwk/dualwaninfo。
func (n *Ntwk) Dualwaninfo() (map[string]interface{}, error) {
	return n.S.Get("ntwk/dualwaninfo", nil, "api")
}

// LanWanConfig 对应 ntwk/lan-wan-config。
func (n *Ntwk) LanWanConfig() (map[string]interface{}, error) {
	return n.S.Get("ntwk/lan-wan-config", nil, "api")
}

File: src/api/ntwk.rs
//! Ntwk API group (`api/Ntwk.py`).

use serde_json::Value;

use crate::connection::Connection;
use crate::errors::Result;

use super::get_value;

/// Ntwk API group.
pub struct Ntwk<'a> {
    conn: &'a Connection,
}

impl<'a> Ntwk<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        Ntwk { conn }
    }

    /// `ntwk/lan_upnp_portmapping`. LAN UPnP port mapping.
    pub fn lan_upnp_portmapping(&self) -> Result<Value> {
        get_value(self.conn, "api/ntwk/lan_upnp_portmapping")
    }

    /// `ntwk/celllock`. Cell lock status.
    pub fn celllock(&self) -> Result<Value> {
        get_value(self.conn, "api/ntwk/celllock")
    }

    /// `ntwk/dualwaninfo`. Dual WAN information.
    pub fn dualwaninfo(&self) -> Result<Value> {
        get_value(self.conn, "api/ntwk/dualwaninfo")
    }

    /// `ntwk/lan-wan-config`. LAN/WAN configuration.
    pub fn lan_wan_config(&self) -> Result<Value> {
        get_value(self.conn, "api/ntwk/lan-wan-config")
    }
}
